Dockerfile

```diff
@@ -0,0 +1,31 @@
+# Use the custom base image with Ruby, Node.js, npm, and Shopify CLI pre-installed
+FROM aaronmedinaaligent/shopify-cli-base:latest
+
+# Set build arguments for user ID and group ID
+ARG USER_ID
+ARG GROUP_ID
+
+# Create a group and user matching the host UID and GID to avoid permission issues
+RUN groupadd -g $GROUP_ID shopifygroup && \
+    useradd -u $USER_ID -g shopifygroup -m shopifyuser
+
+# Create the /theme directory and set the correct permissions
+RUN mkdir -p /shopify && chown shopifyuser:shopifygroup /shopify
+
+# Install xdg-utils for opening URLs
+RUN apt-get update && apt-get install -y xdg-utils
+
+# Modify permissions during the build process
+RUN chown -R shopifyuser:shopifygroup /usr/local/lib/node_modules/@shopify/cli/dist/assets/cli-ruby
```

Comment:
- This could be replaced if the volumes can be mounted instead of recursively chowning files.
- Or COPY them with `chown`

Reply:
The cli-ruby directory is installed as part of the base image (aaronmedinaaligent/shopify-cli-base which will be adjusted later). The chown command is necessary to adjust the ownership of this directory so that developers can run the container with proper permissions under their user accounts when using the environment locally.

This setup ensures that the required files have the correct permissions to avoid access issues when developers work in their local environments.

Reply:
thinking about this again, i feel like we don't need to build a specific image after our base image.

we should also be able to entirely skip creating user and group and pass them as args with docker when launching containers. this is possible both via `docker run` and `compose`.